### PR TITLE
Support Ginkgo v2 only

### PR DESCRIPTION
**Pull request by skitt**
Now that all Submariner projects have switched to Ginkgo v2, we can drop support for v1.

Depends on https://github.com/submariner-io/subctl/pull/466
Depends on https://github.com/submariner-io/coastguard/pull/115
Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

File: scripts/shared/e2e.sh [hunk 1]
```diff
@@ -27,21 +27,12 @@ function generate_kubecontexts() {
...
-        # shellcheck disable=SC2086 # TEST_ARGS is split on purpose
-        ${GO:-go} test -v -timeout 30m -args -test.timeout 15m \
-            -submariner-namespace $SUBM_NS "${clusters[@]/#/-dp-context=}" \
-            -ginkgo.v -ginkgo.randomizeAllSpecs -ginkgo.trace \
-            -ginkgo.reportPassed -ginkgo.reportFile "${DAPPER_OUTPUT}/e2e-junit.xml" \
-            $TEST_ARGS 2>&1 | tee "${DAPPER_OUTPUT}/e2e-tests.log"
-    fi
+    # shellcheck disable=SC2086 # TEST_ARGS is split on purpose
+    ${GO:-go} test -v -timeout 30m -args -test.timeout 15m \
+        -submariner-namespace $SUBM_NS "${clusters[@]/#/-dp-context=}" \
```
Comment (mkolesnik) [inline scripts/shared/e2e.sh]:
```suggestion
    "${GO:-go}" test -v -timeout 30m -args -test.timeout 15m \
        -submariner-namespace "$SUBM_NS" "${clusters[@]/#/-dp-context=}" \
```

These should be quoted though, as we dont want them accidentally splitting.

File: scripts/shared/unit_test.sh [hunk 2]
```diff
@@ -56,13 +56,8 @@ for module in "${modules[@]}"; do
...
         # It's important that the `go test` command's exit status is reported from this () block.
         # Can't be one command (with -cover). Need detailed -coverprofile for Sonar and summary to console.
-        if go mod graph | grep 'submariner.*ginkgo/v2' >/dev/null; then
-            ${GO:-go} test -v ${race} -coverprofile unit.coverprofile "${packages[@]}" --ginkgo.v --ginkgo.trace --ginkgo.junit-report junit.xml "$@" && \
-            go tool cover -func unit.coverprofile
-        else
-            ${GO:-go} test -v ${race} -coverprofile unit.coverprofile "${packages[@]}" -ginkgo.v -ginkgo.trace -ginkgo.reportPassed -ginkgo.reportFile junit.xml "$@" && \
-            go tool cover -func unit.coverprofile
-        fi
+        ${GO:-go} test -v ${race} -coverprofile unit.coverprofile "${packages[@]}" --ginkgo.v --ginkgo.trace --ginkgo.junit-report junit.xml "$@" && \
```
Comment (mkolesnik) [inline scripts/shared/unit_test.sh]:
```suggestion
        "${GO:-go}" test -v ${race} -coverprofile unit.coverprofile "${packages[@]}" --ginkgo.v --ginkgo.trace --ginkgo.junit-report junit.xml "$@" && \
```

Same here

**Review by mkolesnik — CHANGES_REQUESTED** [review]
Few tweaks, otherwise LGTM

**Comment by skitt** [comment]
We can’t do this, Coastguard still uses Ginkgo v1 and can’t migrate to v2 (it relies on old frameworks).

**Review by tpantelis — APPROVED** [review]

**Review by sridhargaddam — APPROVED** [review]

**Comment by tpantelis** [comment]
> We can’t do this, Coastguard still uses Ginkgo v1 and can’t migrate to v2 (it relies on old frameworks).

https://github.com/submariner-io/coastguard/pull/115 switches **coastguard** to v2 so the changes to _unit_test.sh_ can be added back.

**Review by mkolesnik — APPROVED** [review]

**Pull request merged**